doc/nrf/app_bootloaders.rst

```diff
@@ -0,0 +1,74 @@
+.. _app_bootloaders:
+
+Bootloaders and Device Firmware Updates
+#######################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+There are two types of bootloaders used by the |NCS|:
+
+* :ref:`Immutable first-stage bootloaders <immutable_bootloader>`
+* :ref:`Upgradable second-stage bootloaders <upgradable_bootloader>`
+
+These types can be combined for use in products that need a second-stage bootloader.
+A second-stage bootloader adds the capability of providing device firmware updates to itself and the application.
+This is useful when a bootloader performs additional complex operations, like using a software stack.
+
+The bootloaders support two types of updates:
+
+* Direct updates - where an in-place substitution of the image takes place.
+* Background updates - where the updated image is obtained and stored, but the update is completed later on.
+
+You can deliver the updated images to the device in two ways:
+
+* Wired - where updates are sent through a wired connection, like UART, or delivered connecting a flash device.
+* Over-the-air (OTA) where updates are sent through a wireless connection, like Bluetooth LE.
+
+Not every bootloader supported in the |NCS| can be used as either a first-stage or a second-stage one.
+You can find an overview of the bootloaders currently supported and the roles in which they can be used in the table below:
+
+.. _app_bootloaders_support_table:
+
+.. list-table:: Bootloaders supported by |NCS|
+   :widths: auto
+   :header-rows: 1
+
+   * - Bootloader
+     - Can be First-stage
+     - Can be Second-stage
+     - Key Type Support
+     - Public Key Revocation
+     - SMP Updates
```

Comment:
That the bootloader is usable with a DFU transport. AFAIK we currently don't support BLE DFU with only the immutable bootloader present; we have to have MCUboot on board.

Probably because NSIB doesn't integrate mcumgr, right? Would "mcumgr support" be a better indicator for this in the table?

Reply:
That the bootloader can work with the DFU transport on its own. AFAIK we can't do a BLE DFU with just the NSIB; we have to have MCUboot on board.

This is probably because NSIB does not implement mcumgr, right? Would "SMP Updates" be better as "mcumgr support"?